pkg/entry/mongo/document_test.go

```diff
@@ -1,98 +1,127 @@
-			Entry("customer", mongo.CustomerKey, false),
-			Entry("platform ID", mongo.PlatformIDKey, false),
-		)
-	})
+    var ctx context.Context
+
+    BeforeEach(func() {
+        ctx = context.TODO()
+
+        logger, err := log.New(log.OutputPlugin, "test")
+        Expect(err).ToNot(HaveOccurred())
+
+        ctx = log.WithLogger(ctx, logger)
+    })
+
+    Describe("Job with all fields", func() {
+        var entry map[interface{}]interface{}
+
+        BeforeEach(func() {
+            entry = map[interface{}]interface{}{
+                mongo.LogKey:            stringEntry("log"),
+                mongo.StreamKey:         stringEntry("stream"),
+                mongo.TimeKey:           timeEntry(time.Now()),
+                mongo.JobExecutionIDKey: stringEntry("jobExecutionID"),
+                mongo.ProjectIDKey:      stringEntry("projectID"),
+                mongo.CustomerKey:       stringEntry("customer"),
+                mongo.PlatformIDKey:     stringEntry("platformID"),
+            }
+        })
+
+        It("Should work", func() {
+            d, err := mongo.Convert(ctx, time.Now(), entry)
+            Expect(err).ToNot(HaveOccurred())
+            Expect(d).ToNot(BeNil())
+            Expect(d).To(BeAssignableToTypeOf(&mongo.JobLogDocument{}))
+            document := d.(*mongo.JobLogDocument)
+            Expect(document.JobExecutionId).To(BeEquivalentTo(stringEntry("jobExecutionID")))
+            Expect(document.Customer).To(BeEquivalentTo(entry[mongo.CustomerKey]))
+        })
+    })
+
+    Describe("App with all fields", func() {
+        var entry map[interface{}]interface{}
+
+        BeforeEach(func() {
+            entry = map[interface{}]interface{}{
+                mongo.LogKey:            stringEntry("log"),
+                mongo.StreamKey:         stringEntry("stream"),
+                mongo.TimeKey:           timeEntry(time.Now()),
+                mongo.AppExecutionIDKey: stringEntry("appExecutionID"),
+                mongo.AppIDKey:          stringEntry("appID"),
+                mongo.ContainerIDKey:    stringEntry("containerID"),
+                mongo.ProjectIDKey:      stringEntry("projectID"),
+                mongo.CustomerKey:       stringEntry("customer"),
+                mongo.PlatformIDKey:     stringEntry("platformID"),
+            }
+        })
+
+        It("Should work", func() {
+            d, err := mongo.Convert(ctx, time.Now(), entry)
+            Expect(err).ToNot(HaveOccurred())
+            Expect(d).ToNot(BeNil())
+            Expect(d).To(BeAssignableToTypeOf(&mongo.AppLogDocument{}))
+            document := d.(*mongo.AppLogDocument)
+            Expect(document.AppExecutionId).To(BeEquivalentTo(stringEntry("appExecutionID")))
+            Expect(document.ContainerId).To(BeEquivalentTo(stringEntry("containerID")))
+            Expect(document.Customer).To(BeEquivalentTo(entry[mongo.CustomerKey]))
+        })
+    })
+
+    Describe("With missing field", func() {
```

Comment:
Sorry to tell you that it should be `Context()` here.
The description is explicitly `With [a context]`.

Reply:
To explicite : You want 2 Describe when app and job are named byt 1 context here ?

Reply:
Right, that is what I expect :-)